### PR TITLE
SITE-62: Deleting not needed recipe execution

### PR DESCRIPTION
**Pull request by wAsnk**
[SITE-62](https://lombiq.atlassian.net/browse/SITE-62)

[SITE-62]: https://lombiq.atlassian.net/browse/SITE-62?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Themes/Sample.Theme/Recipes/DotNest.Core.SDK.Setup.recipe.json [hunk 1]
```diff
@@ -140,10 +140,6 @@
     {
       "name": "recipes",
       "Values": [
-        {
-          "executionid": "DotNest.Core.SDK.Web",
-          "name": "Lombiq.BaseTheme.LayersAndZones"
-        },
```
Comment (Piedone) [inline src/Themes/Sample.Theme/Recipes/DotNest.Core.SDK.Setup.recipe.json:146]: Shouldn't this be caught be the UI tests?

Comment (wAsnk) [reply src/Themes/Sample.Theme/Recipes/DotNest.Core.SDK.Setup.recipe.json:146]: What UI test?

Comment (wAsnk) [reply src/Themes/Sample.Theme/Recipes/DotNest.Core.SDK.Setup.recipe.json:146]: The SDK doesn't contain UI tests.

Comment (Piedone) [reply src/Themes/Sample.Theme/Recipes/DotNest.Core.SDK.Setup.recipe.json:146]: I've seen this in the **Sites** SDK. It's the same problem, of course, but I think that should have a UI test at least to run the setups or something, because something being fundamentally broken like this can easily be caught like that.

Comment (wAsnk) [reply src/Themes/Sample.Theme/Recipes/DotNest.Core.SDK.Setup.recipe.json:146]: Yes I know. Sites SDK also doesn't have UI tests, which is OK I think, because DotNest covers our Site themes.
This only happens for this specific Sample theme. Do you want a simple UI test in DotNest for this Sample theme?

Comment (Piedone) [reply src/Themes/Sample.Theme/Recipes/DotNest.Core.SDK.Setup.recipe.json:146]: Yeah, let's do that, please. Just a setup and homepage verification. This can BTW be part of this solution, actually, so people can also easily have such basic UI testing for their apps as well (I'd use it for https://github.com/Piedone/DotNest-Sites/).

Comment (wAsnk) [reply src/Themes/Sample.Theme/Recipes/DotNest.Core.SDK.Setup.recipe.json:146]: Okay, I'll add it here.

Comment (Piedone) [reply src/Themes/Sample.Theme/Recipes/DotNest.Core.SDK.Setup.recipe.json:146]: Great, thanks.

File: src/DotNest.Core.SDK.Web/Program.cs [hunk 2]
```diff
@@ -29,3 +30,15 @@
 app.UseStaticFiles();
 app.UseOrchardCore();
 app.Run();
+
+[SuppressMessage(
+    "Design",
+    "CA1050: Declare types in namespaces",
+    Justification = "As described here: https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-6.0.")]
```
Comment (Piedone) [inline src/DotNest.Core.SDK.Web/Program.cs]:
```suggestion
    Justification = "As described here: https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests.")]
```

File: src/Themes/Sample.Theme/Manifest.cs [hunk 3]
```diff
@@ -7,7 +7,7 @@
     Version = "0.0.1",
     Website = "https://github.com/Lombiq/DotNest-Core-SDK",
     Description = "A sample DotNest theme for local development. It must be packaged and used as a Media Theme on DotNest.",
-    BaseTheme = "Lombiq.BaseTheme",
+    BaseTheme = "TheTheme",
```
Comment (Piedone) [inline src/Themes/Sample.Theme/Manifest.cs:10]: Why this change?

File: src/Themes/Sample.Theme/Sample.Theme.csproj [hunk 4]
```diff
@@ -22,11 +22,16 @@
 
   <ItemGroup>
     <Content Include=".config\dotnet-tools.json" />
+    <Content Include="Recipes\*.Tests.*.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+    <Content Include="Views\Home\Index.cshtml" />
+    <Content Include="Views\_ViewImports.cshtml" />
   </ItemGroup>
```
Comment (Piedone) [inline src/Themes/Sample.Theme/Sample.Theme.csproj]: I don't think these are needed.

File: src/Themes/Sample.Theme/Views/Widget-LayoutInjection.liquid
Comment (Piedone) [inline src/Themes/Sample.Theme/Views/Widget-LayoutInjection.liquid:1]: Why did you remove this?

File: src/Themes/Sample.Theme/Sample.Theme.csproj
Comment (Piedone) [inline src/Themes/Sample.Theme/Sample.Theme.csproj:1]: This shows two empty folders:
image

File: src/Themes/Sample.Theme/Views/_ViewImports.cshtml
Comment (Piedone) [inline src/Themes/Sample.Theme/Views/_ViewImports.cshtml:1]: The theme uses only Liquid, this shouldn't be needed.

File: test/DotNest.Core.SDK.Tests.UI/DotNest.Core.SDK.Tests.UI.csproj
Comment (Piedone) [inline test/DotNest.Core.SDK.Tests.UI/DotNest.Core.SDK.Tests.UI.csproj:1]: Add a note about this test project, and how and why you'd have such a project in your own solution.

File: test/DotNest.Core.SDK.Tests.UI/Recipes/DotNest.Core.SDK.Tests.Default.recipe.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "DotNest.Core.SDK.Tests.Default",
+  "displayName": "TEST: DotNest Core SDK - Default tenant",
+  "description": "Setup recipe for the default tenant of DotNest Tenants SDK, for automated UI test execution.",
+  "author": "Lombiq Technologies",
+  "website": "https://dotnest.com/",
```
Comment (Piedone) [inline test/DotNest.Core.SDK.Tests.UI/Recipes/DotNest.Core.SDK.Tests.Default.recipe.json]: Use the same in every recipe. `DotNest.Core.SDK.Setup.recipe` now has a bogus website anyway.

```suggestion
  "website": "https://github.com/Lombiq/DotNest-Core-SDK",
```

File: test/DotNest.Core.SDK.Tests.UI/Recipes/DotNest.Core.SDK.Tests.Default.recipe.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "DotNest.Core.SDK.Tests.Default",
+  "displayName": "TEST: DotNest Core SDK - Default tenant",
+  "description": "Setup recipe for the default tenant of DotNest Tenants SDK, for automated UI test execution.",
+  "author": "Lombiq Technologies",
```
Comment (Piedone) [inline test/DotNest.Core.SDK.Tests.UI/Recipes/DotNest.Core.SDK.Tests.Default.recipe.json:5]: Use the same in the two `Sample.Theme` recipes too.

File: test/DotNest.Core.SDK.Tests.UI/Recipes/Configuration.Tests.recipe.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "Configuration.Tests",
+  "displayName": "Configuration Tests",
+  "description": "Common configuration necessary for UI tests.",
+  "author": "Lombiq Technologies",
+  "website": "https://lombiq.com/",
```
Comment (Piedone) [inline test/DotNest.Core.SDK.Tests.UI/Recipes/Configuration.Tests.recipe.json]: Use the same as the other test.

File: test/DotNest.Core.SDK.Tests.UI/Recipes/Configuration.Tests.recipe.json [hunk 8]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "Configuration.Tests",
+  "displayName": "Configuration Tests",
```
Comment (Piedone) [inline test/DotNest.Core.SDK.Tests.UI/Recipes/Configuration.Tests.recipe.json]: Use the same pattern as the other test.

File: test/DotNest.Core.SDK.Tests.UI/Constants/Configurations.cs [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DotNest.Core.SDK.Tests.UI.Constants;
+
+public static class Configurations
+{
+    // We need this configuration temporary until the HTML validation errors in The Coming Soon theme aren't solved
+    // https://github.com/OrchardCMS/OrchardCore/pull/11243
+    public static readonly Func<OrchardCoreUITestExecutorConfiguration, Task> NoAccessibilityCheckingAndNoHtmlValidationConfiguration =
```
Comment (Piedone) [inline test/DotNest.Core.SDK.Tests.UI/Constants/Configurations.cs]: This isn't used.

File: test/DotNest.Core.SDK.Tests.UI/Constants/Configurations.cs [hunk 10]
```diff
@@ -0,0 +1,60 @@
+using Lombiq.Tests.UI.Extensions;
+using Lombiq.Tests.UI.Services;
+using Shouldly;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DotNest.Core.SDK.Tests.UI.Constants;
+
+public static class Configurations
```
Comment (Piedone) [inline test/DotNest.Core.SDK.Tests.UI/Constants/Configurations.cs]: Are these all really needed?

File: test/DotNest.Core.SDK.Tests.UI/UITestBase.cs [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+        Func<UITestContext, Task<Uri>> setupOperation,
+        Func<OrchardCoreUITestExecutorConfiguration, Task> changeConfigurationAsync) =>
+        base.ExecuteTestAsync(
+            testAsync,
+            browser,
+            setupOperation,
+            async configuration =>
+            {
+                if (changeConfigurationAsync != null) await changeConfigurationAsync(configuration);
+            });
```
Comment (Piedone) [inline test/DotNest.Core.SDK.Tests.UI/UITestBase.cs]: This override doesn't add anything to the base and can be removed.

**Review by Piedone — CHANGES_REQUESTED** [review]

File: src/Themes/Sample.Theme/Views/Widget-LayoutInjection.liquid
Comment (wAsnk) [reply src/Themes/Sample.Theme/Views/Widget-LayoutInjection.liquid:1]: There is another file with the name Widget__LayoutInjection.liquid. It's not necessary to have both.

File: test/DotNest.Core.SDK.Tests.UI/DotNest.Core.SDK.Tests.UI.csproj
Comment (wAsnk) [reply test/DotNest.Core.SDK.Tests.UI/DotNest.Core.SDK.Tests.UI.csproj:1]: Do you mean a readme file?

Comment (wAsnk) [reply test/DotNest.Core.SDK.Tests.UI/DotNest.Core.SDK.Tests.UI.csproj:1]: Or an inline comment in the csproj?

File: src/Themes/Sample.Theme/Manifest.cs (hunk 3 above)
Comment (wAsnk) [reply src/Themes/Sample.Theme/Manifest.cs:10]: First of all Lombiq.BaseTheme didn't add anything visible to the Sample.Theme. Secondly, the UI tests were failing because of some missing title or so after setup. But it is not failing now if I change back to the Lombiq.BaseTheme (This is weird). And finally, it just looks much much better, it is visible at first sight that we are using a base theme.

File: test/DotNest.Core.SDK.Tests.UI/DotNest.Core.SDK.Tests.UI.csproj
Comment (Piedone) [reply test/DotNest.Core.SDK.Tests.UI/DotNest.Core.SDK.Tests.UI.csproj:1]: Readme is probably the best.

File: src/Themes/Sample.Theme/Manifest.cs (hunk 3 above)
Comment (Piedone) [reply src/Themes/Sample.Theme/Manifest.cs:10]: OK then!

File: src/Themes/Sample.Theme/.config/dotnet-tools.json
Comment (Piedone) [inline src/Themes/Sample.Theme/.config/dotnet-tools.json:1]: You previously removed this. Why is it needed again?

File: test/DotNest.Core.SDK.Tests.UI/Recipes/Configuration.Tests.recipe.json [hunk 12]
```diff
@@ -1,9 +1,9 @@
 {
-  "name": "Configuration.Tests",
-  "displayName": "Configuration Tests",
+  "name": "DotNest.Core.SDK.Tests.Configuration.Tests",
```
Comment (Piedone) [inline test/DotNest.Core.SDK.Tests.UI/Recipes/Configuration.Tests.recipe.json]: The file name doesn't match this.

File: test/DotNest.Core.SDK.Tests.UI/Readme.md [hunk 13]
```diff
@@ -0,0 +1,11 @@
+# DotNest Core SDK UI Tests
+
+## Overview
+
+This project contains a sample UI test for the DotNest Core SDK, which ensures that basic functionality is working. You can use it as a starting point for your own UI tests. The project uses [Lombiq.Tests.UI module](https://github.com/Lombiq/UI-Testing-Toolbox/blob/dev/Readme.md) which gives you a base for writing UI tests. For more info please check out the [Lombiq.Tests.UI repository](https://github.com/Lombiq/UI-Testing-Toolbox/blob/dev/Readme.md). For more UI test examples see the [Lombiq.Tests.UI.Samples repository](https://github.com/Lombiq/UI-Testing-Toolbox/blob/dev/Lombiq.Tests.UI.Samples/Readme.md).
+
```
Comment (Piedone) [inline test/DotNest.Core.SDK.Tests.UI/Readme.md]:
```suggestion
This project contains a sample UI test for the DotNest Core SDK, which ensures that basic functionality is working. You can use it as a starting point for your own UI tests. The project uses the [Lombiq UI Testing Toolbox for Orchard Core project](https://github.com/Lombiq/UI-Testing-Toolbox) which gives you a framework for writing UI tests. For more info please check out its documentatoin. For more UI test examples see the [`Lombiq.Tests.UI.Samples` repository](https://github.com/Lombiq/UI-Testing-Toolbox/blob/dev/Lombiq.Tests.UI.Samples/Readme.md).

```

File: test/DotNest.Core.SDK.Tests.UI/Readme.md [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+## Contributing and support
+
+Bug reports, feature requests, comments, questions, code contributions and love letters are warmly welcome. You can send them to us via GitHub issues and pull requests. Please adhere to our [open-source guidelines](https://lombiq.com/open-source-guidelines) while doing so.
+
+This project is developed by [Lombiq Technologies](https://lombiq.com/). Commercial-grade support is available through Lombiq.
```
Comment (Piedone) [inline test/DotNest.Core.SDK.Tests.UI/Readme.md]: The equivalent of this should always be in the root Readme, which it is, so this can be removed.

**Review by Piedone — CHANGES_REQUESTED** [review]

File: src/Themes/Sample.Theme/.config/dotnet-tools.json
Comment (wAsnk) [reply src/Themes/Sample.Theme/.config/dotnet-tools.json:1]: I just restored it.
